SYCL/ESIMD/api/slm_gather_scatter_heavy.cpp

```diff
@@ -0,0 +1,393 @@
+       wg_id++, wg_offset += WG_CHUNK_SIZE) {
+    // iterate over lanes within the work item - should contain consecutive
+    // numbers
+    constexpr unsigned UB1 = is_gather ? VL : WG_SIZE;
+
+    for (unsigned x = 0; x < UB1; x++) {
+      // iterate over work items (ESIMD threads) within each work group
+      constexpr unsigned UB2 = is_gather ? WG_SIZE : VL;
+
+      for (unsigned y = 0; y < UB2; y++) {
+        bool is_lane_masked =
+            MASKED ? (is_gather ? x == MASKED_LANE : y == MASKED_LANE) : false;
+        unsigned off = wg_offset + y * UB1 + x;
+        T real_gold = is_lane_masked ? (T)DEFAULT_VAL : gold;
+        T val = A[off];
+
+        // If lane was masked during gather operation, it may contain any value,
+        // so skip verification.
+        if (!(is_gather && is_lane_masked) && (val != real_gold)) {
+          if (++err_cnt < 41) {
+            std::cout << "  error at " << off << " (wg_id=" << wg_id
+                      << ", x=" << x << ", y=" << y << "): " << val
+                      << " != " << real_gold << " (gold)\n";
+          }
+        }
+        // traversal is done so that each iteration (size/VL total) of this
+        // inner loop peeks into a consecutive datum value:
+        gold++;
+      }
+    }
+  }
+  if (err_cnt > 0) {
+    std::cout << "  pass rate: "
+              << ((float)(size - err_cnt) / (float)size) * 100.0f << "% ("
+              << (size - err_cnt) << "/" << size << ")\n";
+  }
+  return err_cnt == 0;
+}
+
+template <class T, unsigned VL, unsigned STRIDE, MemIODir Dir, TestCase TC>
+bool test_impl(queue q) {
+  size_t size = VL * STRIDE;
+  using KernelType =
+      std::conditional_t<Dir == MEM_GATHER, GatherKernel<T, VL, STRIDE, TC>,
+                         ScatterKernel<T, VL, STRIDE, TC>>;
+  const char *title = KernelType::get_name();
+
+  std::cout << title << " test, T=" << typeid(T).name() << " VL=" << VL
+            << " STRIDE=" << STRIDE << ", TC=" << tc_to_string<TC>();
+  if (is_masked_tc<TC>()) {
+    std::cout << ", masked_lane=" << MaskedLane;
+  }
+  std::cout << "...\n";
+  T *A = new T[size];
+  T *B = new T[size];
+
+  for (unsigned i = 0; i < size; ++i) {
+    A[i] = (T)i;
+    B[i] = (T)DEFAULT_VAL;
+  }
+
+  try {
+    buffer<T, 1> buf_in(A, range<1>(size));
+    buffer<T, 1> buf_out(B, range<1>(size));
+    range<1> glob_range{size / VL};
+
+    auto e = q.submit([&](handler &cgh) {
+      auto acc_in = buf_in.template get_access<access::mode::read_write>(cgh);
+      auto acc_out = buf_out.template get_access<access::mode::read_write>(cgh);
+      constexpr auto WG_SIZE = STRIDE; // for simplicity of the test
+      KernelType kernel(acc_in, acc_out, MaskedLane);
+      cgh.parallel_for(nd_range<1>{glob_range, range<1>(WG_SIZE)}, kernel);
+    });
+    e.wait();
+  } catch (cl::sycl::exception const &e) {
+    std::cout << "SYCL exception caught: " << e.what() << '\n';
+    delete[] A;
+    delete[] B;
+    return e.get_cl_code();
```

Comment:
get_cl_code is obsolete API causing warnings.
```suggestion
    return false;
```

Reply:
good catch!